codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/http/HttpBindingGenerator.kt

```diff
@@ -0,0 +1,383 @@
+
+import software.amazon.smithy.model.knowledge.HttpBinding
+import software.amazon.smithy.model.knowledge.HttpBindingIndex
+import software.amazon.smithy.model.shapes.BlobShape
+import software.amazon.smithy.model.shapes.CollectionShape
+import software.amazon.smithy.model.shapes.DocumentShape
+import software.amazon.smithy.model.shapes.MapShape
+import software.amazon.smithy.model.shapes.MemberShape
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.Shape
+import software.amazon.smithy.model.shapes.StringShape
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.model.shapes.UnionShape
+import software.amazon.smithy.model.traits.EnumTrait
+import software.amazon.smithy.model.traits.MediaTypeTrait
+import software.amazon.smithy.model.traits.TimestampFormatTrait
+import software.amazon.smithy.rust.codegen.rustlang.CargoDependency
+import software.amazon.smithy.rust.codegen.rustlang.RustModule
+import software.amazon.smithy.rust.codegen.rustlang.RustType
+import software.amazon.smithy.rust.codegen.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.rustlang.asType
+import software.amazon.smithy.rust.codegen.rustlang.render
+import software.amazon.smithy.rust.codegen.rustlang.rust
+import software.amazon.smithy.rust.codegen.rustlang.rustBlock
+import software.amazon.smithy.rust.codegen.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.rustlang.stripOuter
+import software.amazon.smithy.rust.codegen.rustlang.withBlock
+import software.amazon.smithy.rust.codegen.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.makeOptional
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpBindingDescriptor
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpLocation
+import software.amazon.smithy.rust.codegen.smithy.protocols.Protocol
+import software.amazon.smithy.rust.codegen.smithy.protocols.parse.EventStreamUnmarshallerGenerator
+import software.amazon.smithy.rust.codegen.smithy.rustType
+import software.amazon.smithy.rust.codegen.util.UNREACHABLE
+import software.amazon.smithy.rust.codegen.util.dq
+import software.amazon.smithy.rust.codegen.util.hasTrait
+import software.amazon.smithy.rust.codegen.util.isPrimitive
+import software.amazon.smithy.rust.codegen.util.isStreaming
+import software.amazon.smithy.rust.codegen.util.toSnakeCase
+
+/**
+ * The type of HTTP message from which we are deserializing the HTTP-bound data.
+ * We are either:
+ *     - deserializing data from an HTTP request (we are a server); or
+ *     - deserializing data from an HTTP response (we are a client).
+ */
+enum class HttpMessageType {
+    REQUEST, RESPONSE
+}
+
+/**
+ * This class generates Rust functions that deserialize data from an HTTP message.
+ * They are useful for *both*:
+ *     - servers (that deserialize HTTP requests); and
```

Comment:
```suggestion
 *     - servers (that deserialize HTTP requests).
```

Reply:
All these are sentences split across bullet points. The style is:

```
* item 1,
* item 2,
* ...
* item n-1; and
* item n.
```

where `and` can of course be any other grammatical conjunction.

Or:

```
* item 1,
* item 2,
* ...
* item n-1; and
* item n

if the sentence continues past the bullet points and ends in the next line.
```

I don't think there's an agreed upon standard, but some formal / technical writing does use this style.

I don't have strong opinions on this, I just try to be consistent.

Reply:
I have no strong opinions as well. It just feels weird to me, but not a blocker. Same for the rest below.